src/shared/components/DurationSelector.tsx

```diff
@@ -33,40 +62,130 @@ const DurationSelector: FunctionComponent<Props> = ({
   )
 
   if (!selected) {
-    selected = {duration: selectedDuration, displayText: selectedDuration}
+    selected = {
+      duration: selectedDuration,
+      displayText: ruleToString(durationToSeconds(selectedDuration)),
+    }
     resolvedDurations = [selected, ...resolvedDurations]
   }
 
+  const defaultCustomDurationValue = 2
+  const defaultCustomDurationUnit = durationUnits[0] // day(s)
+  const [customDurationClicked, setCustomDurationClicked] = useState(false)
+  const [customDurationUnit, setCustomDurationUnit] = useState<string>(
+    defaultCustomDurationUnit
+  )
+  const [customDurationValue, setCustomDurationValue] = useState<number>(
+    defaultCustomDurationValue
+  )
+
+  useEffect(() => {
+    if (customDurationClicked) {
+      onSelectDuration(
+        getCustomDurationString(customDurationValue, customDurationUnit)
+      )
+    }
+  }, [customDurationValue, customDurationUnit, customDurationClicked])
+
+  const getCustomDurationString = (
+    durationValue: number,
+    durationUnit: string
+  ) => {
+    return `${durationValue}${durationUnit}`
+  }
+  const timeInputComponentStyle = {width: '175px'}
+  const customDurationContainerStyle = {marginTop: '16px'}
```

Comment:
I think you'll want to move these out of the function component since they don't rely on anything in the component to work.